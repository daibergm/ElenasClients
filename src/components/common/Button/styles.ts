import { StyleSheet } from 'react-native';

// @Assets
import { Colors } from '../../../assets';

export default StyleSheet.create({
  font: {
    marginTop: 5,
  },
  title: {
    fontSize: 18,
  },
  textSecondary: {
    color: Colors.primary,
  },
  buttonDefault: {
    height: 40,
    borderRadius: 20,
  },
  buttonPrimary: {
    backgroundColor: Colors.primary,
  },
  buttonSecondary: {
    backgroundColor: Colors.white,
    borderColor: Colors.primary,
    borderWidth: 1,
  },
  buttonDisabled: {
    opacity: 0.65,
  },
});
