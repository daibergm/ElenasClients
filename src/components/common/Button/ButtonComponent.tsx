import React from 'react';
import { Button, ButtonProps } from 'react-native-elements';
import { StyleProp, ViewStyle } from 'react-native';

// @Assets
import styles from './styles';

interface Props extends ButtonProps {
  secondary?: boolean;
  danger?: boolean;
}

function ButtonComponent({ secondary, ...otherProps }: Props) {
  let buttonStyle: StyleProp<ViewStyle> = secondary
    ? styles.buttonSecondary
    : styles.buttonPrimary;

  return (
    <Button
      titleStyle={[styles.title, secondary && styles.textSecondary]}
      buttonStyle={[styles.buttonDefault, buttonStyle]}
      disabledStyle={styles.buttonDisabled}
      {...otherProps}
    />
  );
}

export default ButtonComponent;
