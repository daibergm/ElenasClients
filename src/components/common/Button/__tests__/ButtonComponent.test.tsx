import React from 'react';
import { render } from '@testing-library/react-native';

// @Components
import Button from '../ButtonComponent';

describe('Button test suits', () => {
  it('Render correctly', () => {
    const button = render(<Button />);
    expect(button).toMatchSnapshot();
  });
  it('Render as a secondary button', () => {
    const { getByTestId } = render(<Button testID="commonButton" secondary />);
    const button = getByTestId('commonButton');
    expect(button).toBeTruthy();
  });
});
