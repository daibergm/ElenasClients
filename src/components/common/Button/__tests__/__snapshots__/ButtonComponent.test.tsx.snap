// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suits Render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 3,
        "overflow": "hidden",
      },
      Object {
        "borderRadius": 3,
      },
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00adee",
          "borderColor": "#2089dc",
          "borderRadius": 20,
          "borderWidth": 0,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "padding": 8,
        }
      }
    />
  </View>
</View>
`;
